tests/unity/CMakeLists.txt

```diff
@@ -0,0 +1,181 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+zephyr_library()
+
+set_property(GLOBAL PROPERTY CMOCK_DIR ${ZEPHYR_BASE}/../cmock)
+get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+
+find_program(
+  RUBY_EXECUTABLE
+  ruby
+)
+if(${RUBY_EXECUTABLE} STREQUAL RUBY-NOTFOUND)
+    message(FATAL_ERROR "Unable to find ruby")
+endif()
+
+message(STATUS ${RUBY_EXECUTABLE})
+set_property(GLOBAL PROPERTY RUBY_EXECUTABLE ${RUBY_EXECUTABLE})
+
+zephyr_include_directories(
+	${CMOCK_DIR}/vendor/unity/src
+	${CMOCK_DIR}/src
+)
+
+zephyr_library_sources(
+	${CMOCK_DIR}/vendor/unity/src/unity.c
+	${CMOCK_DIR}/src/cmock.c
+)
+
+# Generate test runner file.
+function(test_runner_generate test_file_path)
+    get_property(RUBY_EXECUTABLE GLOBAL PROPERTY RUBY_EXECUTABLE)
```

Comment:
find_program sets it's output parameter as a CACHE variable, which always has global scope, so we don't need to wrap the ruby executable in a property.

Reply:
done